app/assets/javascripts/application.js

```diff
@@ -13,7 +13,31 @@
 //= require jquery
 //= require jquery_ujs
 //= require angular/angular
+//= require angular-route/angular-route
+//= require angular-rails-templates
 //= require_tree .
 
 
-pinterest = angular.module("pinterest", []);
+pinterest = angular.module("pinterest", ["templates", "ngRoute", "controllers"]);
+
+// TODO: move me to router
+pinterest.config(["$routeProvider", function($routeProvider) {
+  $routeProvider.when("/", {
+    templateUrl: "index.html",
+    controller: "ProductsController"
+  });
+}]);
+
+controllers = angular.module("controllers", []);
+
+// TODO: move me to another file
+controllers.controller("ProductsController", ["$scope", function($scope) {
+  $scope.products = products;
+}]);
+
+// TODO: move to server
+products = [
+  { id: 1, blurb: "blurb 1", details_url: "", title: "Prod 1", thumbnail_url: "" },
+  { id: 2, blurb: "blurb 2", details_url: "", title: "Prod 2", thumbnail_url: "" },
```

Comment:
Line is too long.<br>Identifier 'details_url' is not in camel case.<br>Identifier 'thumbnail_url' is not in camel case.